### PR TITLE
Version 2.2

### PR DESCRIPTION
**Pull request by richerarc**
This PR Adds the following:

- Operation ID under the operation feature
- Middleware chain consume and return a Context instead of a response
- Responders can now access the HttpContext
- Server ID can be configured

### CODE REVIEW

File: saphir/src/http_context.rs [hunk 1]
```diff
@@ -1,19 +1,412 @@
...
+    /// something else
+    pub fn take_request(&mut self) -> Option<Request> {
+        match std::mem::take(self) {
+            State::Before(r) => Some(r),
+            _ => None,
+        }
+    }
+
+    /// Take the current request leaving `State::Empty` behind
+    /// Panics if the state is not `Before`
```
Comment (LucFauvel) [inline saphir/src/http_context.rs]: per the rust api doc guideline, the panics line should be under the panics section like this `/// # Panics`
see https://rust-lang.github.io/api-guidelines/documentation.html

File: saphir/src/http_context.rs [hunk 2]
```diff
@@ -1,19 +1,412 @@
...
+    /// something else
+    pub fn take_response(&mut self) -> Option<Response> {
+        match std::mem::take(self) {
+            State::After(r) => Some(r),
+            _ => None,
+        }
+    }
+
+    /// Take the current response leaving `State::Empty` behind
+    /// Panics if the state is not `After`
```
Comment (LucFauvel) [inline saphir/src/http_context.rs]: Same as other panics

File: saphir/src/http_context.rs (hunk 1 above)
Comment (richerarc) [reply saphir/src/http_context.rs]: done

File: saphir/src/http_context.rs (hunk 2 above)
Comment (richerarc) [reply saphir/src/http_context.rs]: done

**Pull request merged**